src/arrayMethodReduce.js

```diff
@@ -6,6 +6,19 @@
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
     // write code here
+    let previous = initialValue;
+    let start = 0;
+
+    if (arguments.length < 2) {
```

Comment:
this is not an obvious check. Let's use more appropriate condition here:
```suggestion
    if (initialValue === undefined) {
```